django/contrib/sites/models.py

```diff
@@ -37,10 +38,21 @@ def _get_site_by_id(self, site_id):
 
     def _get_site_by_request(self, request):
         host = request.get_host()
-        if host not in SITE_CACHE:
-            site = self.get(domain__iexact=host)
-            SITE_CACHE[host] = site
-        return SITE_CACHE[host]
+
+        # First attempt to look up the site by host with or without port:
+        try:
+            if host not in SITE_CACHE:
+                SITE_CACHE[host] = self.get(domain__iexact=host)
+            return SITE_CACHE[host]
+
+        # Fallback to looking up site after stripping port from the host:
```

Comment:
Will also move the comment inside the ``try`` as it makes more sense with the same indentation.